python/_server.py

```diff
@@ -0,0 +1,252 @@
+from enum import Enum, IntEnum
+from collections.abc import Iterable, Mapping, Tuple
+from typing import Any, Union
+from ._infer import *
+from ._trace import *
+
+
+class ModelControlMode(Enum):
+    NONE: int = 0
+    POLL: int = 1
+    EXPLICIT: int = 2
+
+
+# Use IntEnum for flags that may be multiplexed
+class BatchProperty(IntEnum):
+    UNKNOWN: int = 1
+    FIRST_DIM: int = 2
+
+
+class TransactionProperty(IntEnum):
+    ONE_TO_ONE: int = 1
+    DECOUPLED: int = 2
+
+
+class ModelIndexFlag(IntEnum):
+    READY: int = 1
+
+
+class RateLimiterOptions:
+
+    class Mode(Enum):
+        OFF: int = 0
+        EXECUTION_COUNT: int = 1
+
+    class Resource:
+
+        def __init__(self, name: str, count: int, device_id: int) -> None:
+            self._name = name
+            self._count = count
+            self._device_id = device_id
+
+    def __init__(self, mode: Mode = None) -> None:
+        self._mode = mode
+        self._resources: Iterable[RateLimiterOptions.Resource] = None
+        pass
+
+    def add_resource(self, name: str, count: int, device_id: int) -> None:
+        if self._resources is None:
+            self._resources = []
+        self._resources.append(
+            RateLimiterOptions.Resource(name, count, device_id))
+
+
+class Options:
+    # in-process API doesn't provide proper getter,
+    # can't reflect the default / current value.
+    # NOTE: TRITONSERVER_ServerOptions creation will be in place
+    # during server initialization
+    def __init__(self) -> None:
+        # List the set of options, 'None' implies that the value is not
+        # specified and the default value will be used
+        self.server_id: str = None
+        self.minimum_compute_capability: float = None
+        self.strict_readiness: bool = None
+        self.exit_on_error: bool = None
+        self.exit_timeout: int = None
+
+        # [FIXME] this option is actually not in used in core
+        # TRITONSERVER_ServerOptionsSetBufferManagerThreadCount
+        self.buffer_manager_thread_count: int = None
+
+        # Model stuff..
+        self.model_repositories: Iterable[str] = None
```

Comment:
I was wondering if it would make sense to set the model repository path as a required field.

Reply:
As in making the paths part of init parameter? i.e. `__init__(self, model_paths)`

Reply:
Yes. I just thought that instead of doing this
```
options = Options()
options.model_repositories = ["/path/to/models"]
```
we can do
```
options = Options(["/path/to/models"])
```
given that the model repo paths are required when starting the server.